mopidy_spotify/playlists.py

```diff
@@ -65,16 +144,114 @@ def refresh(self):
         self._loaded = True
 
     def create(self, name):
-        pass  # TODO
+        logger.info(f"Creating playlist {name}")
+        uri = web.create_playlist(self._backend._web_client, name)
+        self._get_flattened_playlist_refs()
+        return self.lookup(uri) if uri else None
 
     def delete(self, uri):
-        pass  # TODO
+        logger.info(f"Deleting playlist {uri}")
+        ok = web.delete_playlist(self._backend._web_client, uri)
+        self._get_flattened_playlist_refs()
+        return ok
+
+    @staticmethod
+    def _len_replace(playlist, n=_chunk_size):
+        return math.ceil(len(playlist.tracks) / n)
+
+    @staticmethod
+    def _is_spotify_track(track_uri):
+        return track_uri.startswith("spotify:track:")
 
     def save(self, playlist):
-        pass  # TODO
+        saved_playlist = self._get_playlist(playlist.uri, with_owner=True)
+        if not saved_playlist:
+            return
+
+        saved_playlist, owner = saved_playlist
+        # We limit playlist editing to the user's own playlists, since mopidy
+        # mangles the names of other people's playlists.
+        if owner and owner != self._backend._web_client.user_id:
+            logger.error(
+                f"Refusing to modify someone else's playlist ({playlist.uri})"
+            )
+            return
+
+        # We cannot add or (easily) remove spotify:local: tracks, so refuse
+        # editing if the current playlist contains such tracks.
+        if any(
+            (t.uri.startswith("spotify:local:") for t in saved_playlist.tracks)
```

Comment:
I've made a _is_spotify_local. adding it to web.linktype/web.weblink doesn't really work, since local tracks don't have ids. and modifying the class attributes to properly represent local tracks when we will never read them seems unnecessary.